edith-step/src/components/ReceiptInput.js

```diff
@@ -0,0 +1,245 @@
+   * @param {Props} props Setup state.
+   */
+  constructor(props) {
+    super(props);
+    this.state = {items: [], itemName: '', itemPrice: 0.0, itemQuantity: 1,
+                  itemCategory: 'category', itemReceiptId: 'receiptId'};
+    this.getDate = this.getDate.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  getDate() {
+      const date = new Date(Date.now());
+      let month = date.getMonth() + 1;
+      month < 10 ? month = "0" + month.toString() : month = month.toString();
+      let day = date.getDate();
+      day < 10 ? day = "0" + day.toString() : day = day.toString();
+      let year = date.getFullYear();
+      return [year, month, day].join("-");   
+  }
+
+  /**
+   * Sends item info to the receipt-data servlet and
+   * responds with the cheapest store for that item.
+   *
+   * @param {String} name item name
+   * @param {double} price item price
+   * @param {number} quantity item quantity
+   * @return {Object} newDeal cheapest item
+   */
+  async getDeal(name, price, quantity) {
+    const response = await axios({
+      method: 'post',
+      url: '/receipt-data',
+      data: {
+        itemName: name,
+        itemPrice: price,
+        itemQuantity: quantity,
+      },
+    });
+    // TODO: Implement enum.
+    const dealItem = response.data;
+    console.log(dealItem);
+
+    const newDeal = dealItem === 'NO_STORE' ?
+      {storeName: 'No deal found.', storePrice: 0} :
+      {storeName: dealItem.storeName, storePrice: dealItem.price};
+
+    return newDeal;
+  }
+
+  /**
+   * Calls the getDeal function and passes it data on a particular
+   * item (name, price, quantity). The helper function returns the cheapest
+   * store for the item passed, and the items list in the state is updated
+   * to include the deal.
+   * @param {Event} e Submission event.
+   */
+  async handleSubmit(e) {
+    e.preventDefault();
+    if (this.state.itemName.length === 0) {
+      return;
+    }
+
+    const newDeal = await this.getDeal(this.state.itemName,
+        this.state.itemPrice, this.state.itemQuantity);
+
+    const dealMessage = newDeal.storeName == 'NO_STORE' ||
+      newDeal.storePrice > this.state.itemPrice ?
+      'No deal found.' :
+      `Purchase at ${newDeal.storeName} for $${newDeal.storePrice}.`;
+
+    const newItem = {
+      itemName: this.state.itemName,
+      itemPrice: this.state.itemPrice,
+      itemQuantity: this.state.itemQuantity,
+      itemDeal: dealMessage,
+      id: Date.now(),
+    };
+
+    axios({
+      method: 'post',
+      url: '/user-stats-servlet',
+      data: {
+        itemName: this.state.itemName,
+        itemCategory: this.state.itemCategory,
+        itemPrice: this.state.itemPrice,
+        itemQuantity: this.state.itemQuantity,
+        itemDate: this.getDate(),
+        itemReceiptId: this.state.itemReceiptId
+      }
+    }).then((response) => {
+      console.log(response);
```

Comment:
is the response needed for anything? or could it be a catch instead of a then

Reply:
I changed it to a catch